hadoop-ozone/interface-storage/src/main/java/org/apache/hadoop/ozone/om/OMMetadataManager.java

```diff
@@ -544,4 +544,10 @@ String getMultipartKey(long volumeId, long bucketId,
    */
   long getBucketId(String volume, String bucket) throws IOException;
 
+  /**
+   * Returns List<{@link BlockGroup}> for a key in the deletedTable.
+   * @param deletedKey - key to be purged from the deletedTable
+   * @return {@link BlockGroup}
+   */
+  List<BlockGroup> getBlocksForKeyDelete(String deletedKey) throws IOException;
```

Comment:
We shouldn't limit it, otherwise we don't reclaim all blocks of a particular key. This is similar to KeyDeletingService which also only limits on number of keys. Also `ozone.snapshot.key.deleting.limit.per.task` ensures that we have a hard limit on the number of keys we process.

Reply:
Got it. Make sense. Thought this was for a group of keys.